test/distributed/org/apache/cassandra/distributed/test/TestBaseImpl.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test;
+
+import com.datastax.driver.core.ResultSet;
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.ICluster;
+import org.apache.cassandra.distributed.api.IInstance;
+import org.apache.cassandra.distributed.impl.RowUtil;
+import org.apache.cassandra.distributed.shared.Builder;
+import org.apache.cassandra.distributed.shared.DistributedTestBase;
+
+public class TestBaseImpl extends DistributedTestBase
```

Comment:
I don't see why we need this class.  I assume its because of `Cluster.build()` but even then thats a api thing and should fix the API rather than try to hide this.

Can we remove this class?

Reply:
In other repo you've asked to remove dependency of `DistributedTestBase` on junit, which I did. Now, however, we need `TestBaseImpl` for before/after hooks. Also, if you take a look at the test runner repo, you'll see that we do fake dependency injection using this class, relying on its stable API, to be able to run tests for all other repos.